### ISSUE TITLE
TreeGrid - add support for Template and Component renderers for the hierarchy column

### ISSUE THREAD

**Issue by gilberto-torrezan** [vaadin/vaadin-grid-flow#431]
Currently the hierarchy column in TreeGrid can only be added with a `ValueProvider` - it's not possible to customize the component being rendered at all.

```java
public Column<T> addHierarchyColumn(ValueProvider<T, ?> valueProvider) {
```

It's true that the template needs to be encapsulated by the `vaadin-grid-tree-toggle` webcomponent, but that shouldn't prevent the usage of custom templates (and components) inside it.

**Comment by enver-haase** [comment]
We need this too please.

**Comment by mvysny** [comment]
What is the target version for this please?

**Comment by tulioag** [comment]
The fix is already released in 4.2.0.alpha1 and 5.1.0.alpha1 of vaadin-grid-flow.
It will be available in platforms 14.3 and 16.
